soundfile.py

```diff
@@ -216,9 +238,59 @@
     __libsndfile_version__ = __libsndfile_version__[len('libsndfile-'):]
 
 
-def read(file, frames=-1, start=0, stop=None, dtype='float64', always_2d=False,
-         fill_value=None, out=None, samplerate=None, channels=None,
-         format=None, subtype=None, endian=None, closefd=True):
+
+@overload
+def read(file: FileDescriptorOrPath, frames: int = -1, start: int = 0, stop: Optional[int] = None, 
+         *, dtype: Literal['float64'] = 'float64', always_2d: bool = False,
+         fill_value: Optional[float] = None, out: None = None, 
+         samplerate: Optional[int] = None, channels: Optional[int] = None,
+         format: Optional[Format] = None, subtype: Optional[SubType] = None, 
+         endian: Optional[Endian] = None, closefd: bool = True) -> Tuple[npt.NDArray[np.float64], int]:
+    ...
+
+@overload
+def read(file: FileDescriptorOrPath, frames: int = -1, start: int = 0, stop: Optional[int] = None, 
+         *, dtype: Literal['float32'], always_2d: bool = False,
+         fill_value: Optional[float] = None, out: None = None, 
+         samplerate: Optional[int] = None, channels: Optional[int] = None,
+         format: Optional[Format] = None, subtype: Optional[SubType] = None, 
+         endian: Optional[Endian] = None, closefd: bool = True) -> Tuple[npt.NDArray[np.float32], int]:
+    ...
+
+@overload
+def read(file: FileDescriptorOrPath, frames: int = -1, start: int = 0, stop: Optional[int] = None, 
+         *, dtype: Literal['int32'], always_2d: bool = False,
+         fill_value: Optional[float] = None, out: None = None, 
+         samplerate: Optional[int] = None, channels: Optional[int] = None,
+         format: Optional[Format] = None, subtype: Optional[SubType] = None, 
+         endian: Optional[Endian] = None, closefd: bool = True) -> Tuple[npt.NDArray[np.int32], int]:
+    ...
+
+@overload
+def read(file: FileDescriptorOrPath, frames: int = -1, start: int = 0, stop: Optional[int] = None, 
+         *, dtype: Literal['int16'], always_2d: bool = False,
+         fill_value: Optional[float] = None, out: None = None, 
+         samplerate: Optional[int] = None, channels: Optional[int] = None,
+         format: Optional[Format] = None, subtype: Optional[SubType] = None, 
+         endian: Optional[Endian] = None, closefd: bool = True) -> Tuple[npt.NDArray[np.int16], int]:
+    ...
+
+@overload
+def read(file: FileDescriptorOrPath,  frames: int = -1, start: int = 0, stop: Optional[int] = None, 
+         dtype: Dtype = 'float64', always_2d: bool = False,
+         fill_value: Optional[float] = None, *, out: T_ndarr, 
+         samplerate: Optional[int] = None, channels: Optional[int] = None,
+         format: Optional[Format] = None, subtype: Optional[SubType] = None, 
+         endian: Optional[Endian] = None, closefd: bool = True) -> Tuple[T_ndarr, int]:
+    ...
+
+def read(file: FileDescriptorOrPath, frames: int = -1, start: int = 0, stop: Optional[int] = None, 
```

Comment:
why was this reformatted with spaces around the equal sign?

Reply:
I fixed this comment, but please consider to use code formatter tool like `black`, `ruff`.

https://github.com/bastibe/python-soundfile/pull/462/commits/29c215fb43d8b7640f8634f68586ef07dbb79294